pyomo/contrib/fbbt/expression_bounds_walker.py

```diff
@@ -221,20 +261,20 @@ def _handle_named_expression(visitor, node, arg):
 }
 
 
-_operator_dispatcher = ExitNodeDispatcher(
-    {
-        ProductExpression: _handle_ProductExpression,
-        DivisionExpression: _handle_DivisionExpression,
-        PowExpression: _handle_PowExpression,
-        AbsExpression: _handle_AbsExpression,
-        SumExpression: _handle_SumExpression,
-        MonomialTermExpression: _handle_ProductExpression,
-        NegationExpression: _handle_NegationExpression,
-        UnaryFunctionExpression: _handle_UnaryFunctionExpression,
-        LinearExpression: _handle_SumExpression,
-        Expression: _handle_named_expression,
-    }
-)
+class ExpressionBoundsExitNodeDispatcher(ExitNodeDispatcher):
+    def unexpected_expression_type(self, visitor, node, *args):
+        if isinstance(node, NumericExpression):
+            ans = -inf, inf
+        elif isinstance(node, BooleanExpression):
+            ans = Bool(False), Bool(True)
+        else:
+            super().unexpected_expression_type(visitor, node, *args)
+        logger.warning(
```

Comment:
Will this issue a warning for external functions? If so, I think that needs changed.

Reply:
It will not: ExternalFunctions are a) caught in the `beforeChild` callback (so `exitNode` should never be called, and b) there is an explicit registration for ExternalFunctionExpression, so it will never hit the `unexpected_xpression_type` method.